src/index.d.ts

```diff
@@ -44,6 +42,28 @@ export interface PromiseState<T = {}> {
   ) => PromiseStateLike<T> | PromiseStateLike<TRejected>;
 }
 
+export interface PendingPromiseState<T = {}> extends PromiseStateBase {
+  readonly pending: true;
+  readonly fulfilled: false;
+  readonly rejected: false;
+}
+
+export interface FulfilledPromiseState<T = {}> extends PromiseStateBase {
```

Comment:
would it make sense to change Line 18 to:

```tsx
resolve<T = {}>(value?: PromiseStateLike<T>, meta?: any): FulfilledPromiseState<T>;
```

and likewise for the other static methods? Otherwise, those will all be `PendingPromiseState<T>`, right?

I'm wondering how this is going to impact the static `all` and `race` methods. I'm kind of wondering if we need to still have a generic type for those...

Reply:
Yes, that's much better! I also update other static methods excepts `all` and `race`. I think it's not so much helpful to introduce generic interfaces for these 2 methods.